pkg/render/node.go

```diff
@@ -604,24 +648,38 @@ func (c *nodeComponent) getCalicoIPAM() string {
 	} else {
 		assign_ipv6 = "false"
 	}
-	return fmt.Sprintf(`{ "type": "calico-ipam", "assign_ipv4" : "%s", "assign_ipv6" : "%s"}`,
-		assign_ipv4, assign_ipv6,
-	)
+	return map[string]interface{}{
+		"type":        "calico-ipam",
+		"assign_ipv4": assign_ipv4,
+		"assign_ipv6": assign_ipv6,
+	}
 }
 
-func buildHostLocalIPAM(cns *operatorv1.CalicoNetworkSpec) string {
+func buildHostLocalIPAM(cns *operatorv1.CalicoNetworkSpec) map[string]interface{} {
 	v6 := GetIPv6Pool(cns.IPPools) != nil
 	v4 := GetIPv4Pool(cns.IPPools) != nil
 
 	if v4 && v6 {
 		// Dual-stack
-		return `{ "type": "host-local", "ranges": [[{"subnet": "usePodCidr"}],[{"subnet": "usePodCidrIPv6"}]]}`
+		return map[string]interface{}{
+			"type": "host-local",
+			"ranges": [][]map[string]string{
+				[]map[string]string{{"subnet": "usePodCidr"}},
```

Comment:
These inner type definitions are actually redundant and not needed in go - you can do this instead:

```go
return map[string]interface{}{
		"type":   "host-local",
		"ranges": [][]map[string]string{{{"subnet": "usePodCidr"}}, {{"subnet": "usePodCidrIPv6"}}},
	}
```

Go knows how to infer the type of the list elements from the main `[][]map[string]string`

Reply:
Done.